flatland/envs/agent_utils.py

```diff
@@ -67,11 +64,64 @@ def load_env_agent(agent_tuple: Agent):
 
 @attrs
 class EnvAgent:
+    # backwards compatibility
+    # TODO drop setters, keep only getters for backwards compatibility but get core free of setters
```

Comment:
TODO drop setters, keep only getters for backwards compatibility but get core free of setters

Reply:
TODO (1'): make EnvAgent typed on configuration type?